GVFS/GVFS.Virtualization/FileSystemCallbacks.cs

```diff
@@ -630,7 +636,15 @@ private FileSystemTaskResult ExecuteBackgroundOperation(FileSystemTask gitUpdate
 
                 case FileSystemTask.OperationType.OnFileDeleted:
                     metadata.Add("virtualPath", gitUpdate.VirtualPath);
-                    result = this.AddModifiedPathAndRemoveFromPlaceholderList(gitUpdate.VirtualPath);
+                    if (this.newPaths.Contains(gitUpdate.VirtualPath))

```

Comment:
Two related comments:

1. Currently `OnFileDeleted` is scheduled from the PreDelete callback on Mac.  However, on Mac we can't remove files from ModifiedPaths in PreDelete because we don't actually know that the file has been deleted.

2. Related to 1 (and something @jrbriggs and I were just discussing), should we try to design this to work on mac and Windows from the start?  Or at this point should we just ensure what whatever we do for Windows does not break Mac?  

For this specific feature it might only take a few tweaks to get things working on both platforms (e.g. on Mac we could take an extra step of checking if the file actually was deleted before calling `TryRemoveModifiedPath`).  Thoughts?

CC: @sanoursa for his thoughts as well.

Reply:
> However, on Mac we can't remove files from ModifiedPaths in PreDelete because we don't actually know that the file has been deleted.

The bare minimum change for this PR could be to wrap the new code here in an IsUnderConstruction check.

There's still the open question though of whether we design\implement for both platforms from the start.

Reply:
I think it would be nice to start thinking about how to make all new features cross-platform from the start. Unless we're up against a part of the Mac design that is still unknown or in heavy flux, I would take the time now to make this work for both.